packages/server/src/manifest-builder.ts

```diff
@@ -24,14 +24,33 @@ interface ManifestBuilderOptions {
   distDir: string;
 };
 
+function ftruncate(fd: number, len: number): Operation {
+  return ({ fail, resume }) => {
+    fs.ftruncate(fd, len, err => {
+      if (err) {
+        fail(err);
+        return;
+      }
+      resume();
+    })
+  };
+}
+
+// https://github.com/nodejs/node/issues/34189#issuecomment-654878715
+function* truncate(path: string, len: number): Operation {
+  let file: fs.promises.FileHandle = yield open(path, 'r+');
+  yield ensure(() => file.close());
```

Comment:
We're still on the fence about when the right time is to use `ensure()`, since in the cases where it's necessary, it's almost always modeled better using a resource. In this case, a simple try/finally is "just javascript" and communicates the intent clearly to all javascript programmers.

```ts
let file: fs.promises.FileHandle = yield open(path, 'r+');
try {
  yield ftruncate(file.fd, len);
} finally {
  file.close();
}